servers/services/src/main/java/org/projectnessie/services/rest/BaseResource.java

```diff
@@ -87,17 +78,11 @@ protected BaseResource(
       }
 
       // we need to make sure that the hash in fact exists on the named ref
-      Hash hash = Hash.of(hashOnRef);
-      try (Stream<WithHash<CommitMeta>> commits = store.getCommits(namedRefWithHash.getValue())) {
-        if (commits.noneMatch(c -> c.getHash().equals(hash))) {
-          throw new NessieNotFoundException(
-              String.format("Could not find commit '%s' in reference '%s'.", hashOnRef, namedRef));
-        }
-      }
-      return WithHash.of(hash, namedRefWithHash.getValue());
+      return WithHash.of(
```

Comment:
I think, it makes sense to compare `hashOnRef` w/ `refWithHash.getHash()` and just return `namedRefWithHash` when equal to prevent the then useless call to `getStore().hashOnReference()` (could be placed into the above `if`

Reply:
Would probably require some inline-doc to explain when the if-condition is met.

Reply:
yep that makes sense, implemented it and added a comment explaining why we can exit early